### DISCUSSION TITLE
Visualization with EVE-NG or Pnetlab

### DISCUSSION THREAD

**Discussion by bgp-router** [ipspace/netlab#1258] [Ideas]
Dear netlab team,

I'm delivering trainings for network, security, storage, Linux and Windows using Pnetlab.
I'm looking for an automation tool to build new labs on KVM and docker.

Do you have any plans to make lab visualization with EVE-NG or Pnetlab?

**Comment by ipspace** [comment]
Hi there!

Thanks for the question. We have no plans for further platform integration (like EVE-NG or Pnetlab). They both seem to be a self-contained VM with their own lab provisioning system and UI, so it's hard to see what the added value might be.

You can obviously always generate device configurations with netlab, or even install netlab on the virtual machine running EVE-NG or Pnetlab (at least EVE-NG seems to be running on Ubuntu).

If you would work out the integration details, for example what configuration file to generate to platform X (similarly to what we do for containerlab and Vagrant), then it should be relatively easy to add a provider module for that platform.

Hope this helps, Ivan

**Comment by castrojuanj** [comment]
> Pnetlab

Never ever download or pay for Pnetlab (is a pirate version of EVE-NG). You only have to visit their website and you figure out is a phony software (it could work) but is suspicious the give you cisco and juniper images. Also look at their mail , is a gmail mail.

**Comment by bgp-router** [comment]
Thank you for your warning!

I'm using CML2 now. This has an API, Ansible and Terraform support, so I want to use AI agents to setup labs and play Red Team during the cyber crisis exercises.
